.github/workflows/deploy.yml

```diff
@@ -1,32 +1,56 @@
+# Sample workflow for building and deploying a mdBook site to GitHub Pages
+#
+# To get started with mdBook see: https://rust-lang.github.io/mdBook/index.html
+#
 name: CI
-on: [push, pull_request]
+on:
+  workflow_dispatch:
+  push:
+    tags: 
+      - '*-publish'
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
 
 jobs:
+  # Build job
   build:
-    name: Build
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-      with:
-        repo-token: ${{ secrets.GITHUB_TOKEN }}
-    - name: configure git
-      run: |
-        git config --global user.name "SichangHe"
-        git config --global user.email "sichanghe@users.noreply.github.com"
-    - name: pull mdbook build
-      run: git clone https://github.com/SichangHe/mdbook_binaries.git
-    - name: add bin to path
-      run: echo "$(pwd)/mdbook_binaries/bin" >> $GITHUB_PATH
-    - name: Download KaTeX CSS and fonts
-      run: python3 download_static_css.py
-    - name: Build
-      run: mdbook build
-    - name: push
-      run: |
-        git fetch
-        git switch gh-pages
-        rm -rf docs
-        mv book docs
-        git add docs
-        git commit --allow-empty -m "Deploy $GITHUB_SHA to gh-pages"
-        git push -f
+      - uses: actions/checkout@v3
+      - name: Install mdBook
+        run: |
+          curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh
+          rustup update
+          cargo install mdbook
+          cargo install mdbook-katex
```

Comment:
Please do not install Rust on the runner and compile the two binaries. It takes over 10min in practice and makes the CI unbearably slow.